### PR TITLE
Fix/eslint standarjs conflicts on generator functions

### LINKED ISSUE #1522: Eslint and Standard JS conflicts about generator functions
# 🐛Bug Report 🐛
I am currently using ESlint plugin in my editor (WebStorm). 

Standard JS requires spacing before and after `*` of a generator function.  
image

The space will be removed during pre-commit phase (`lint-staged`) and then editor will show lint error again. It becomes cyclic.

### PR DESCRIPTION
**Pull request by lzhuor**
## Description
Standard JS requires spacing before and after `*` of a generator function.  
image

We could fix this error using auto fix with ESLint's editor plugin (I'm on WebStorm). However, the space will be removed during pre-commit phase (`lint-staged`) and then editor will show lint error again. It becomes cyclic.

## Change Type
Please enter one or more of the following:
- Other (eslint configuration)

## Testing Steps
1. Fix spacing before and after `*` for generator functions
2. Commit and push code
3. There should be no errors about `generator-star-spacing`. The spacing added in step 1 should remain in code. 

## Code Checklist
- [x] Code compiles successfully (verified via `yarn start`)
- [x] `README.md` and other documentation is updated as needed

Fix #1522

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -61,6 +61,7 @@
     "react/self-closing-comp": 2,
     "react/sort-comp": 2,
     "space-before-function-paren": 2,
-    "strict": 0
+    "strict": 0,
+    "generator-star-spacing": ["error", {"before": true, "after": true}]
```
Comment (schnogz) [inline .eslintrc]: While you're in here, could you sort all rules alphabetically? Looks like most of them already are.

**Comment by schnogz** [comment]
Also run `yarn prettier && yarn lint:fix` to apply rules to all files and commit those changes as well.

File: .eslintrc (hunk 1 above)
Comment (lzhuor) [reply .eslintrc]: Good catch. Done. 
Thanks!

**Comment by lzhuor** [comment]
@schnogz thanks! Addressed comments and pushed them.

**Comment by schnogz** [comment]
i see we have a warning when running lint tasks: `Warning: React version not specified in eslint-plugin-react settings. See https://github.com/yannickcr/eslint-plugin-react#configuration .`

Can you add a settings object to `.eslintrc` to fix this warning. It should the following and the warning will go away.

```
"settings": {
    "react": {
      "version": "latest"
    }
  },
```

**Comment by lzhuor** [comment]
@schnogz , Could you please share which command generates that warning? Is it on circleci or local? It seems fine on my end. I have tried `yarn lint`, `yarn lint:fix` and `yarn:frontend`.

**Comment by lzhuor** [comment]
@schnogz Added. Thanks!

**Pull request merged**